src/structs.rs

```diff
@@ -61,132 +61,120 @@ impl Keyword {
+        } else {
+            Self::Terminating { length: value }
+        }
+    }
+}
+
 /// A `Block` defines a fixed-size block of bytes.
 ///
-/// It is used to store variable length values in the Chain Table. It allows
-/// padding lines to an equal size which avoids leaking information.
+/// It is used to store variable length values in the Chain Table by fixed-sized
+/// chunks. It allows padding lines to an equal size which avoids leaking
+/// information.
 ///
 /// A value can be represented by several blocks. The first blocks representing
-/// a value are full, i.e. no padding is necessary. The first byte is set to
-/// `LENGTH`. The last block representing a value may not be full. It is
-/// padded with 0s. The first byte is used to write the size of the data stored
-/// in this block.
+/// a value are full. The first byte is set to `u8::MAX`. The last block
+/// representing a value may not be full. It is padded with 0s. The first byte
+/// is used to write the size of the data stored in this block. The special
+/// prefix value `0` means the block contains no data.
 ///
 /// +-------------------------+--------------------------+
-/// |        1 byte           | `BLOCK_LENGTH` - 1 bytes |
+/// |        1 byte           |      `BLOCK_LENGTH`      |
 /// +-------------------------+--------------------------+
-/// |         `LENGTH`        |                          |
+/// |      `0` or `u8::MAX`   |                          |
 /// |            or           |       padded data        |
 /// | number of bytes written |                          |
 /// +-------------------------+--------------------------+
 ///
-/// The corollary is that the size of a block shall not be greater than 255
-/// (`u8::MAX`). This is enough in practice because bigger blocks imply more
-/// wasted space for small values (a small value would be padded to `LENGTH -
-/// 1`).
+/// The corollary is that the length of a block shall not be greater than 254
+/// (`u8::MAX - 1`); this is enough because bigger blocks imply more wasted
+/// space for small values (a small value would be padded to `LENGTH` bytes).
 #[must_use]
-#[derive(Clone, Debug, Hash, PartialEq, Eq)]
-pub struct Block<const LENGTH: usize>(pub(crate) [u8; LENGTH]);
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub struct Block<const LENGTH: usize> {
+    pub(crate) block_type: BlockType,
+    pub(crate) prefix: BlockPrefix,
+    pub(crate) data: [u8; LENGTH],
+}
 
-impl_byte_array!(Block);
+/// A block length should be a valid `u8` to allow writing in one byte, and the
+/// value `u8::MAX` is reserved to indicate a non-terminating block.
+const MAX_BLOCK_LENGTH: usize = 254; // `u8::MAX - 1`
 
 impl<const LENGTH: usize> Block<LENGTH> {
+    /// This checks that the `BLOCK_LENGTH` does not exceed the
+    /// `MAX_BLOCK_LENGTH`.
+    pub const CHECK_LENGTH: () = assert!(
+        LENGTH <= MAX_BLOCK_LENGTH,
+        "`BLOCK_LENGTH` should be *not* be greater than 254",
```

Comment:
Do not hardcode 254 here?

Reply:
It is not possible to get a constant call to assert.

Reply:
You can put "`BLOCK_LENGTH` should be *not* be greater than `MAX_BLOCK_LENGTH`"?

Reply:
I think a user getting this error might want to know the numerical value rather than the name of the constant.

Reply:
As you wish, nobody will see this error anyway except developers :-D And `BLOCK_LENGTH` is already not shown in the error